scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/GetJobContentFactory.java

```diff
@@ -0,0 +1,153 @@
+        return String.format(FOUT_SPACES_INDENT + "<%s %s=\"%s\" %s=\"%s\"/>",
+                             XMLTags.COMMON_INFO,
+                             XMLAttributes.COMMON_NAME,
+                             pair.getKey(),
+                             XMLAttributes.COMMON_VALUE,
+                             pair.getValue());
+    }
+
+    private String replaceTagContent(String jobContent, XMLTags tag, String newContent, int end) {
+        final String firstHalf = jobContent.substring(0, end);
+
+        /*
+         * because tasks can contain variables as well we modify original string
+         * only till the provided end
+         */
+        final String untouchablePart = jobContent.substring(end);
+
+        final Optional<Matcher> openMatcher = indexOfPattern(firstHalf, tag.getOpenTagPattern());
+        if (!openMatcher.isPresent()) {
+            return jobContent;
+        }
+        final int afterOpenTag = openMatcher.get().end();
+
+        final Optional<Matcher> closeMatcher = indexOfPattern(firstHalf, tag.getCloseTagPattern());
+        if (!closeMatcher.isPresent()) {
+            return jobContent;
+        }
+        final int beforeCloseTag = closeMatcher.get().start();
+
+        final String beforeNewContent = firstHalf.substring(0, afterOpenTag) + System.lineSeparator();
+        final String afterNewContent = TWO_SPACES_INDENT + firstHalf.substring(beforeCloseTag);
+
+        return beforeNewContent + newContent + afterNewContent + untouchablePart;
+    }
+
+    private Optional<Matcher> indexOfPattern(String content, String pattern) {
+        Pattern aPattern = Pattern.compile(pattern);
+        Matcher matcher = aPattern.matcher(content);
+
+        if (matcher.find()) {
+            return Optional.of(matcher);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private String newVariablesContent(Map<String, JobVariable> variables) {
+        return variables.values()
+                        .stream()
+                        .map(this::variableContent)
+                        .map(s -> s + System.lineSeparator())
+                        .reduce("", String::concat);
+    }
+
+    private String variableContent(JobVariable jobVariable) {
+        if (jobVariable.getModel() != null && !jobVariable.getModel().trim().isEmpty()) {
+            return String.format(FOUT_SPACES_INDENT + "<%s %s=\"%s\" %s=\"%s\" %s=\"%s\" />",
+                                 XMLTags.VARIABLE.getXMLName(),
+                                 XMLAttributes.VARIABLE_NAME,
+                                 jobVariable.getName(),
+                                 XMLAttributes.VARIABLE_VALUE,
+                                 jobVariable.getValue(),
+                                 XMLAttributes.VARIABLE_MODEL,
+                                 jobVariable.getModel());
+        } else {
+            return String.format(FOUT_SPACES_INDENT + "<%s %s=\"%s\" %s=\"%s\" />",
```

Comment:
Are this `"<%s %s=\"%s\" %s=\"%s\" />"` and this `"<%s %s=\"%s\" %s=\"%s\"/>"` the same? Maybe you can put it inside a constant.